html/semantics/links/hyperlink-auditing/headers.optional.html

```diff
@@ -0,0 +1,55 @@
+<!DOCTYPE html>
+<meta charset="utf-8">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/common/utils.js"></script>
+
+<iframe id="i" src="/common/blank.html"></iframe>
+
+<!-- Test is optional because hyperlink auditing is optional. -->
+
+<script>
+promise_test(async t => {
+  await new Promise(resolve => window.onload = resolve);
+
+  const id = token();
+
+  const el = document.createElement("a");
+  el.ping = new URL(`resources/stash-headers.py?id=${id}`, location.href); // this will be a POST
+  el.href = "/common/blank.html?1";
+
+  i.contentDocument.body.append(el);
+  el.click();
+
+  let headers;
+  await pollForConditionFunc(t, async () => {
```

Comment:
You can use https://web-platform-tests.org/writing-tests/testharness-api.html#Test.step_wait I think.

Reply:
You cannot sadly because it doesn't work with async funcs.

Reply:
I'll trust you, but it returns a promise so that seems weird.